src/init/CMakeLists.txt

```diff
@@ -1,5 +1,17 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
+if(CONFIG_SOF)  ### Zephyr ###
+
+
+zephyr_library_sources(
```

Comment:
This is a spot where Zephyr's promiscuous cmake layer hurts pretty badly, because you can't add files to the project in the same way in both modes.  Again, maybe work around this by adding yet another layer of indirection and have a "sof_sources()".

Also, FWIW:  the "_library_sources" variant really doesn't do anything useful beyond make the build slower by taking the time to create a static library.  There is (AFAIK) exactly one spot in Zephyr where we have to do this for obscure linkage reasons[1], but for general C code it's almost always faster and cleaner to link all the object files at once.

I've tried to make that point upstream and basically failed: "zephyr_library_sources()" is a pervasive paradigm.  But really it's needless except in specific situations.

(For reference: the reason to use it is that a static library only presents internal files to the link if one of their symbols is accessed, so you can build it and include static stuff like e.g. SYS_INIT() or interrupt declarations, and avoid having those end up in the final output unless something is using the file.  But DTS and kconfig are the right mechanisms for that kind of dependency tracking, doing it with the linker is dangerous and surprising.)

Reply:
Heh, I've spent my bikeshed quota for the month already.

The way this goes is that someone says "libraries make linking faster!" and the response is "only after the ranlib stage builds the table, and in zephyr most symbols are actually needed by the final link", then someone else demands benchmarks, etc...

Really I was mostly just whining about a personal peeve.  I've made my peace with it: per-directory/per-app fine grained static libraries are the way zephyr links itself.